### DISCUSSION TITLE
Reacting to form value change

### DISCUSSION THREAD

**Discussion by bombillazo** [react-hook-form/react-hook-form#8610] [Q&A]
Hello, what property should I use in a `useEffect` hook to check if my form values have changed?

**Comment by inomn** [comment]
Hi! Probably this is what you looking for https://react-hook-form.com/api/useform/watch see examples

**Reply by bombillazo** [reply]
Thanks for the reply, Ill prefase by saying I'm using React Native. I tried using watch like this:

```ts
const { watch } = useFormContext();
const values = watch();

useEffect(()=> {
  // my effect logic
},[values])
```
Unfortunately this yield infinite rerenders and the app reaches a max depth error.

**Reply by bluebill1049** [reply]
I made a video on watch API: https://react-hook-form.com/api/useform/watch watch is made for `render` phase not `useEffect`, do you need useEffect for this? or you can achieve the same within the `onChange` or even potentially without the useEffect

**Reply by bombillazo** [reply]
Hello, thanks for the response. I ultimately need a way to sync the form state to a parent component, so I was thinking of using a state setter function passed as a prop to the child component and syncing using `useEffect` when the form values change.

I cannot move the form to the parent component under the current design
